sda/host.go

```diff
@@ -25,54 +18,29 @@ type Host struct {
 	ServerIdentity *network.ServerIdentity
 	// Our private-key
 	private abstract.Scalar
-	// The TCPHost
-	host network.SecureHost
-	Dispatcher
+	suite   abstract.Suite
+	Router
 	// Overlay handles the mapping from tree and entityList to ServerIdentity.
 	// It uses tokens to represent an unique ProtocolInstance in the system
 	overlay *Overlay
 	// The open connections
 	connections map[network.ServerIdentityID]network.SecureConn
-	// chan of received messages - testmode
-	networkChan chan network.Packet
-	// The suite used for this Host
-	suite abstract.Suite
-	// We're about to close
-	isClosing  bool
-	closingMut sync.Mutex
-	// lock associated to access network connections
-	networkLock sync.RWMutex
 	// lock associated to access trees
-	treesLock sync.Mutex
-	// working address is mostly for debugging purposes so we know what address
-	// is known as right now
-	workingAddress string
-	// listening is a flag to tell whether this host is listening or not
-	listening bool
-	// whether processMessages has started
-	processMessagesStarted bool
-	// tell processMessages to quit
-	ProcessMessagesQuit chan bool
-
+	treesLock            sync.Mutex
 	serviceManager       *serviceManager
 	statusReporterStruct *statusReporterStruct
 }
 
-// NewHost starts a new Host that will listen on the network for incoming
-// messages. It will store the private-key.
+// NewHost returns a new Host that out of a private-key and its relating public
+// key with a default TcpRouter as Router.
```

Comment:
I know that we use the public "everywhere". But I'm obviously missing sth. here: `pkey` stands for private key here, right? (at least it is a `abstract.Scalar`) I think the comment is a little confusing because we pass a private-key and no corresponding public key. In fact, the public key is implicitly passed through the `ServerIdentity`.